component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -728,12 +728,7 @@ private boolean isValidTokenFederatedUser(int token, AuthenticationContext conte
 
         String secretKey = null;
         if (context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL) != null) {
-            try {
-                secretKey = TOTPUtil.decrypt(context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL).
-                        toString());
-            } catch (CryptoException e) {
-                throw new TOTPException("Error while decrypting the secret key", e);
-            }
+            secretKey = context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL).toString();
```

Comment:
No need to check this whether encrypted or not ?

Reply:
Add EnableEncryption claim property check temporarily to avoid issues during the time of migration.